File: sample-projects/python/de/sots/cellarsandcentaurs/domain/model/centaur.py
from .creature import Creature
from .creature_id import CreatureId
from .creature_type import CreatureType


class Centaur(Creature):
    def __init__(self, creature_id: CreatureId):
        super().__init__(creature_id, CreatureType.MONSTROSITY)

    def attack_bonus(self) -> int:
        return super().attack_bonus() + 1

File: sample-projects/javascript/src/de/sots/cellarsandcentaurs/domain/model/Centaur.js
import { Creature } from "./Creature.js";
import { CreatureType } from "./CreatureType.js";
import { SpeedType } from "./SpeedType.js";

export class Centaur extends Creature {
    constructor(id) {
        super(id, CreatureType.MONSTROSITY);
    }

    gallop() {
        const walkingSpeed = this.getSpeeds().get(SpeedType.WALKING);
        return walkingSpeed.getSpeed() * 2;
    }
}
